src/datatype.c

```diff
@@ -951,12 +951,9 @@ JL_DLLEXPORT jl_value_t *jl_new_structv(jl_datatype_t *type, jl_value_t **args,
     if (type->instance != NULL)
         return type->instance;
     jl_value_t *jv = jl_gc_alloc(ptls, jl_datatype_size(type), type);
-    JL_GC_PUSH1(&jv);
     for (size_t i = 0; i < na; i++) {
         set_nth_field(type, (void*)jv, i, args[i]);
     }
-    init_struct_tail(type, jv, na);
```

Comment:
Isn't this still needed?

Reply:
Yes, sorry this got dropped in rebase. I have it fixed locally, but didn't push the version yet.